### DISCUSSION TITLE
Usage Case Forecasts

### DISCUSSION THREAD

**Discussion by Steviey** [TDAmeritrade/stumpy#331] [Help: Coding & Implementations]
I'm wondering if it makes sense to use motif data of stumpy, to annotate a time series and use this as explanatory variables for forecasts. My guess is, there would be a problem with the index-length of a profile, because of the lag of items for the last window. Is there any code example out there (R or Python)?

**Comment by seanlaw** [comment]
@Steviey Thanks for the question and welcome to the STUMPY community. I haven't really come across anything significant related to forecasting with matrix profiles and STUMPY is probably not the best tool for this (specialized tools like [Facebook Prophet](https://facebook.github.io/prophet/) might be more well suited). Having said that, [time series chains](https://stumpy.readthedocs.io/en/latest/Tutorial_Time_Series_Chains.html) is a very crude approach but your mileage may vary. Again, I think there are other better options and, in R, I believe that Rob Hyndman's [forecast package](https://cran.r-project.org/web/packages/forecast/forecast.pdf) is the gold standard.

**Reply by Steviey** [reply]
Hello Sean, Thank you for your welcome. I have all the mentioned R-forcasting packages already in use. What I want to do, is time series annotation with 'domain knowledge' of stumpy. The main forecasting package would be still a specialized one- like prophet, forecast etc. They then would use the annotated time series, to do there work. So the main question would be, **how to annotate the time series with stumpy accordingly.** Simply some more columns to the time series(es). This way- we also could use ML-algos for regressions.

**Reply by seanlaw** [reply]
I haven't thought about this a ton but there are two potentially related concepts that might be useful (just thinking out loud here):

* [Using Weakly Labeled Time Series to Predict Outcomes](https://www.cs.ucr.edu/~eamonn/WeaklyLabeledTimeSeries.pdf)
* [Shapelet Discovery](https://github.com/TDAmeritrade/stumpy/blob/master/docs/Tutorial_Shapelet_Discovery.ipynb)

I will have to think about this more.
